pkg/discovery/dns/resolver_test.go

```diff
@@ -85,25 +85,33 @@ var (
 			resolver: &mockHostnameResolver{
 				resultSRVs: map[string][]*net.SRV{
 					"_test._tcp.mycompany.com": {
-						&net.SRV{Target: "192.168.0.1", Port: 8080},
-						&net.SRV{Target: "192.168.0.2", Port: 8081},
+						&net.SRV{Target: "alt1.mycompany.com.", Port: 8080},
+						&net.SRV{Target: "alt2.mycompany.com.", Port: 8081},
 					},
 				},
+				resultIPs: map[string][]net.IPAddr{
+					"alt1.mycompany.com.": {net.IPAddr{IP: net.ParseIP("192.168.0.1")}},
+					"alt2.mycompany.com.": {net.IPAddr{IP: net.ParseIP("192.168.0.2")}},
+				},
 			},
 		},
 		{
-			testName:       "multiple srv records from srv lookup",
+			testName:       "multiple srv records from srv lookup with specified port",
 			addr:           "_test._tcp.mycompany.com:8082",
 			qtype:          SRV,
```

Comment:
Can we maybe add `FullSRV` type for this kind of lookup? or `NoASRV` for old logic?